### PR TITLE
Output type is "number" instead of "BigNumber" when bits <= 48

### LINKED ISSUE #142: [ethers] uint8 returned data is infered as BigNumber, but ethers return number instead
For constant methods (at least), uint8 numbers (at least) are returned as numbers by ethers.js Contract, but TypeChain generate bindings as if they were `BigNumber`.

### PR DESCRIPTION
**Pull request by andrevmatos**
Fix #142

### CODE REVIEW

**Comment by krzkaczor** [comment]
Thank you for your contribution! 👍 

Could you add a test proving that it works? Take a look at DumbContract.sol. You might want to add there repro case (just a function returning uint8 for example). And then you can write new test case in `test/integration/targets/ethers/DumbContract.spec.ts` Thanks!

**Comment by krzkaczor** [comment]
@andrevmatos i think you need to update snapshots. Try `SNAPSHOT_UPDATE=1 yarn test`

**Comment by andrevmatos** [comment]
Done.

File: __snapshots__/DumbContract.spec.ts.js [hunk 1]
```diff
@@ -230,6 +239,10 @@ export class DumbContract extends TC.TypeChainContract {
     return TC.promisify(this.rawWeb3Contract.byteArray, []);
   }
 
+  public get returnSmallUint(): Promise<BigNumber> {
```
Comment (krzkaczor) [inline __snapshots__/DumbContract.spec.ts.js]: note: these snapshots are used by different target thats why it's BigNumber here

**Comment by krzkaczor** [comment]
@andrevmatos Thanks! Great work! 🥇

**Comment by krzkaczor** [comment]
Released as `typechain@0.3.13`

**Pull request merged**